akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -303,16 +303,16 @@ private[akka] abstract class InternalProjectionState[Offset, Envelope](
     val handlerLifecycle = handlerStrategy.lifecycle
     statusObserver.started(projectionId)
 
-    val source: Source[ProjectionContextImpl[Offset, Envelope], NotUsed] =
+    val source: Source[ProjectionContextImpl[Offset, Envelope], Future[Done]] =
```

Comment:
I agree that it communicates better. 

This was more a clean-up action. This source had already a materialized value as `Future` and we were discarding it on line 329.

And if we map it to `Future[NotUsed]`?

Reply:
Hmm, but actually we don't really care about it because it starts with the `Source.futureSource`. This is creating a `Source[T, Future[M]]`. That `M` is already a `NotUsed`, but we loose this info because of type inference. It becomes an `Any`.

Only the last `Future[Done]` is relevant. The one we create when the source terminates.

Reply:
I revert it to `NotUsed`. Btw, this is not rebased as well.